### PR TITLE
Backport feature consistency for sqlite

### PR DESCRIPTION
**Pull request by ThetaSinner**
### Summary

I've really tried to minimise the possibility of a break here. I think that the `hdi` and the `hdk` will behave the same as they did which is the main thing. People *may* have problems building `holochain` with a partial upgrade of its dependencies but I'm struggling to construct a problem scenario in my head.

I do think this would be worse testing with a pre-release version before it goes into `0.1.5` though

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [x] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/kitsune_p2p/bin_data/Cargo.toml
Comment (neonphog) [inline crates/kitsune_p2p/bin_data/Cargo.toml:1]: delete

File: crates/kitsune_p2p/block/Cargo.toml
Comment (neonphog) [inline crates/kitsune_p2p/block/Cargo.toml:1]: delete

**Review by neonphog — CHANGES_REQUESTED** [review]
You've got some errant files from main. Otherwise looks good.

**Comment by ThetaSinner** [comment]
@neonphog good spots thank you! Always forget that 'empty' in a cherry-pick does not mean 'removed'

**Review by jost-s — APPROVED** [review]

**Pull request merged**